pygin/pygin/src/panel.cpp

```diff
@@ -0,0 +1,61 @@
+#include "headers.hpp"
+
+#include "panel.h"
+#include "far_api.hpp"
+#include "py.integer.hpp"
+
+PluginPanelItem*
+get_find_data::get_items(py::object const& find_data, size_t& item_count, OPERATION_MODES& mode)
+{
+	find_data.ensure_type(far_api::type("GetFindDataInfo"sv));
+	auto py_items = py::cast<py::list>(find_data["PanelItems"sv]);
+	auto size = py_items.size();
+	std::unique_ptr<char[]> mem(new char[sizeof(get_find_data)
+		+ sizeof(PluginPanelItem) * size]);
+	auto* result = reinterpret_cast<PluginPanelItem*>(
+		mem.get() + sizeof(get_find_data));
+	new (mem.get()) get_find_data(py_items, result);
```

Comment:
Why do we need placement new here?

Reply:
Because we are allocating a single memory chunk to hold both the C-array of `PluginPanelItem` and C++ object `get_find_data`. Since Far API operates with `PluginPanelItem*` in terms of allocations/deallocation this is a way to associate C++ object with `PluginPanelItem*` to destroy C++ later having only `PluginPanelItem*`.

Reply:
I see. Yes, API doesn't really shine here. We might need some wrapper later to make it tidier.